src/flint/types/fmpz_mod.pxd

```diff
@@ -1,14 +1,19 @@
 from flint.flint_base.flint_base cimport flint_scalar
 from flint.flintlib.fmpz cimport fmpz_t
-from flint.flintlib.fmpz_mod cimport fmpz_mod_ctx_t
+from flint.flintlib.fmpz_mod cimport (
+    fmpz_mod_ctx_t,
+    fmpz_mod_discrete_log_pohlig_hellman_t
+)
 
 
 cdef class fmpz_mod_ctx:
     cdef fmpz_mod_ctx_t val
-
+    cdef fmpz_mod_discrete_log_pohlig_hellman_t L
+    cdef bint _dlog_precomputed
+    cdef any_as_fmpz_mod(self, obj)
+    cdef _precompute_dlog_prime(self)
```

Comment:
In memory terms the first element of L is an `fmpz_mod_ctx_t` which would end up being a copy of `val`. If we include the struct `L` inline here then we don't need `val`.

I think it is better though to just have a pointer like
```
cdef fmpz_mod_discrete_log_pohlig_hellman_t *L
```
If the pointer is set to NULL in `__cinit__` then there is no need for a separate `_dlog_computed` variable. The `_precompute_dlog_prime` method can just check for NULL and then call malloc etc.

Reply:
I have not addressed this yet. So you think I should set the pointer to `NULL` for `__cinit__` and then when I do the normal discrete log stuff then I want to work with `&self.ctx.L`? 

I guess I can also do something similar for `x_g` with a pointer?

Reply:
I think it's like `f(L[0])`. You can't use `f(*L)` because that means something else in Python.

Reply:
Ahh got it! The real issue was my `malloc` which was poorly written before.

```py
self.L = <fmpz_mod_discrete_log_pohlig_hellman_t *>libc.stdlib.malloc(1 * cython.sizeof(fmpz_mod_discrete_log_pohlig_hellman_struct))
```

Seems to work

Reply:
(There was no `flint_malloc` which I could see, so I did it this way, but we could maybe write a small function to do this?)